lib/libp2p_port.ex

```diff
@@ -688,4 +729,85 @@ defmodule LambdaEthereumConsensus.Libp2pPort do
       add_subscriber(state, topic, module)
     end)
   end
+
+  defp on_tick(time, %{genesis_time: genesis_time} = state) when time < genesis_time, do: state
+
+  defp on_tick(time, %{genesis_time: genesis_time, slot_data: slot_data} = state) do
+    # TODO: we probably want to remove this (ForkChoice.on_tick) from here, but we keep it
+    # here to have this serialized with respect to the other fork choice store modifications.
+
+    ForkChoice.on_tick(time)
+
+    new_slot_data = compute_slot(genesis_time, time)
+
+    updated_state = maybe_tick_validators(slot_data != new_slot_data, new_slot_data, state)
+
+    maybe_log_new_slot(slot_data, new_slot_data)
+
+    updated_state
+  end
+
+  defp maybe_tick_validators(false = _slot_data_changed, _slot_data, state), do: state
+
+  defp maybe_tick_validators(true, slot_data, %{validators: validators} = state) do
+    updated_validators = notify_validators(validators, {:on_tick, slot_data})
+
+    %{state | slot_data: slot_data, validators: updated_validators}
+  end
+
+  defp notify_validators(validators, msg) do
+    start_time = System.monotonic_time(:millisecond)
+
+    Logger.debug("[Libp2p] Notifying all Validators with message: #{inspect(msg)}")
+
+    updated_validators = Enum.map(validators, &notify_validator(&1, msg))
+
+    end_time = System.monotonic_time(:millisecond)
+
+    Logger.debug(
+      "[Validator Manager] #{inspect(msg)} notified to all Validators after #{end_time - start_time} ms"
+    )
+
+    updated_validators
+  end
+
+  defp notify_validator({pubkey, validator}, {:on_tick, slot_data}),
+    do: {pubkey, Validator.handle_tick(slot_data, validator)}
+
+  defp notify_validator({pubkey, validator}, {:new_block, slot, head_root}),
```

Comment:
Maybe we can have all of this stuff in a Validator module?

Reply:
I was moving this to a new module that represents the state of the pool of validators, but agree that we could just move it to the Validator module in the meantime (thats on me for renaming the ValidatorManger xD), I'll move them!

Reply:
Moved it again to `Validator.Setup`, to remove it from libp2p, it will be better handled in the follow-up PR.